docs/versioned_docs/version-v3.0.0/03-Features/01-Azure/01-service-bus.mdx

```diff
@@ -0,0 +1,371 @@
+
+:::tip
+Use the [`ManagedIdentityCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.managedidentitycredential) where possible.
+:::
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+Host.CreateDefaultBuilder()
+    .ConfigureServices(services =>
+    {
+        // Register the message pump via a `TokenCredential` implementation.
+        // 🔗 See Microsoft's guidance on Azure authentication: 
+        // https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme
+        services.AddServiceBusQueueMessagePump(
+            "<queue-name>",
+            "<fully-qualified-namespace>",
+            new ManagedIdentityCredential());
+
+
+        // Register the message pump via a custom implementation factory, returning a `ServiceBusClient`.
+        // 🔗 See Microsoft's guidance on Azure core extensions to re-use clients in your application:
+        // https://learn.microsoft.com/en-us/dotnet/api/overview/azure/microsoft.extensions.azure-readme
+        services.AddServiceBusTopicMessagePump(
+            "<topic-name>",
+            "<subscription-name>",
+            "<fully-qualified-namespace>",
+            (IServiceProvider services) =>
+            {
+                return services.GetRequiredService<ServiceBusClient>();
+            });
+    });
+```
+
+### Register your message handlers on message pump
+Your custom *message handlers* can be registered on a *message pump* registration. Both the type of the handler and the type of messages it can process, is required.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+Host.CreateDefaultBuilder()
+    .ConfigureServices(services =>
+    {
+        // highlight-start
+        services.AddServiceBusTopicMessagePump(...)
+                .WithServiceBusMessageHandler<OrderMessageHandler, Order>()
+                .WithServiceBusMessageHandler<OrderV2MessageHandler, OrderV2>();
+        // highlight-end
+    });
+```
+
+:::warning[Considerations]
+* The order in which the *message handlers* are registered matters when a message is routed.
+* Only the first matching *message handler* will be used to process the message, even when multiple match.
+* Multiple *message handlers* with the same (message) type can registered, but they need to distinguish themselves with [routing options](#message-handler-routing-customization).
+:::
+
+## Customization
+Due to the wide range of situations within messaging solutions, Arcus Messaging supports highly customizable message pump/handler registrations.
+
+### Message pump customization
+The following options are available when registering the Azure Service Bus message pump.
+
+```csharp
+services.AddServiceBus[Topic/Queue]MessagePump(..., options => 
+{
+    // The amount of concurrent calls to process messages 
+    // (default: null, leading to the defaults of the Azure Service Bus SDK message handler options).
+    options.MaxConcurrentCalls = 5;
```

Comment:
We've created this property with this name, but it's actuallly misleading.  We've an issue to fix this property name.  In this documentation, I'd change the comment on L127 to something that says that this property defines how many messages the message-pump will retrieve in one call.  (We're not doing multiple concurrent calls to retrieve messages).

Reply:
Or we also include the issue in v3 so we can already mention this new property.

Reply:
PR added https://github.com/arcus-azure/arcus.messaging/pull/673, will change the docs as well.

Reply:
Updated the docs here: https://github.com/arcus-azure/arcus.messaging/pull/671/commits/79a3cf2a8767480ce36b1e77ed58aa0b76cbd7f0